src/plugins/saved_objects_management/public/management_section/objects_table/components/duplicate_modal.tsx

```diff
@@ -0,0 +1,467 @@
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      allSelectedObjects: this.props.selectedSavedObjects,
+      workspaceOptions: [],
+      allWorkspaceOptions: [],
+      targetWorkspaceOption: [],
+      isLoading: false,
+      isIncludeReferencesDeepChecked: true,
+      savedObjectTypeInfoMap: new Map<string, [number, boolean]>(),
+    };
+  }
+
+  workspaceToOption = (
+    workspace: WorkspaceAttribute,
+    currentWorkspaceName?: string
+  ): WorkspaceOption => {
+    // add (current) after current workspace name
+    let workspaceName = workspace.name;
+    if (workspace.name === currentWorkspaceName) {
+      workspaceName += ' (current)';
+    }
+    return {
+      label: workspaceName,
+      key: workspace.id,
+      value: workspace,
+    };
+  };
+
+  async componentDidMount() {
+    const { workspaces } = this.props;
+    const currentWorkspace = workspaces.currentWorkspace$.value;
+    const currentWorkspaceName = currentWorkspace?.name;
+    const targetWorkspaces = this.getTargetWorkspaces();
+
+    // current workspace is the first option
+    const workspaceOptions = [
+      ...(currentWorkspace ? [this.workspaceToOption(currentWorkspace, currentWorkspaceName)] : []),
+      ...targetWorkspaces
+        .filter((workspace: WorkspaceAttribute) => workspace.name !== currentWorkspaceName)
+        .map((workspace: WorkspaceAttribute) =>
+          this.workspaceToOption(workspace, currentWorkspaceName)
+        ),
+    ];
+
+    this.setState({
+      workspaceOptions,
+      allWorkspaceOptions: workspaceOptions,
+    });
+
+    const { duplicateMode } = this.props;
+    if (duplicateMode === DuplicateMode.All) {
+      const { allSelectedObjects } = this.state;
+      const categorizedObjects = groupBy(allSelectedObjects, (object) => object.type);
+      const savedObjectTypeInfoMap = new Map<string, [number, boolean]>();
+      for (const [savedObjectType, savedObjects] of Object.entries(categorizedObjects)) {
+        savedObjectTypeInfoMap.set(savedObjectType, [savedObjects.length, true]);
+      }
+      this.setState({ savedObjectTypeInfoMap });
+    }
+
+    this.isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this.isMounted = false;
+  }
+
+  getTargetWorkspaces = () => {
+    const { workspaces } = this.props;
+    const workspaceList = workspaces.workspaceList$.value;
+    return workspaceList.filter((workspace) => !workspace.libraryReadonly);
+  };
+
+  duplicateSavedObjects = async (savedObjects: SavedObjectWithMetadata[]) => {
+    this.setState({
+      isLoading: true,
+    });
+
+    const targetWorkspace = this.state.targetWorkspaceOption[0].key;
+
+    await this.props.onDuplicate(
+      savedObjects,
+      this.state.isIncludeReferencesDeepChecked,
+      targetWorkspace!
+    );
+
+    if (this.isMounted) {
+      this.setState({
+        isLoading: false,
+      });
+    }
+  };
+
+  onSearchWorkspaceChange = (searchValue: string) => {
```

Comment:
I think ComboBox has default behvior of filtering by the search value, can we use that instead of filtering by ourself?

Reply:
Yes, I will delete the useless search logic.